### ISSUE TITLE
Crash upon connecting with JS enabled

### ISSUE THREAD

**Issue by VmKid** [Freeyourgadget/Gadgetbridge#863]
As documented in #858, GB crashes on some devices when attempting to connect with Experimental JS enabled.

In my case, my devices are an AT&T Galaxy S5 (SM-G900A) running Android 6.0.1 connecting to a Pebble Time.

A full logcat of GB events is located [here](https://pastebin.com/4JhGRW29). 

Of note is what appears to be a trace:
```
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler: Uncaught exception: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is th...
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.app.ContextImpl.startActivity(ContextImpl.java:738) ~[na:0.0]                                                          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.app.ContextImpl.startActivity(ContextImpl.java:725) ~[na:0.0]                                                          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at android.content.ContextWrapper.startActivity(ContextWrapper.java:345) ~[na:0.0]                                                
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.connect(PebbleIoThread.java:170) ~[na:0.0]          
10-29 14:04:46.206 20413 14123 E nodomain.freeyourgadget.gadgetbridge.LoggingExceptionHandler:  at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.run(PebbleIoThread.java:197) ~[na:0.0]              
10-29 14:04:46.206 20413 14123 E AndroidRuntime: Process: nodomain.freeyourgadget.gadgetbridge, PID: 20413                                                                                                                        
10-29 14:04:46.206 20413 14123 E AndroidRuntime:    at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.connect(PebbleIoThread.java:170)                                                                  
10-29 14:04:46.206 20413 14123 E AndroidRuntime:    at nodomain.freeyourgadget.gadgetbridge.service.devices.pebble.PebbleIoThread.run(PebbleIoThread.java:197)                                                                      
10-29 14:04:46.216   811  1670 W ActivityManager:   Force finishing activity nodomain.freeyourgadget.gadgetbridge/.activities.ControlCenterv2                                                                                    
10-29 14:04:46.316   811   905 D ISSUE_DEBUG: InputChannelName : 9294e15 Application Error: nodomain.freeyourgadget.gadgetbridge                                                                                                  
```

Which appears to refer to [this line, launching the JS intent](https://github.com/Freeyourgadget/Gadgetbridge/blob/e028b4d22380a19fb6e9dade689e4151f29dedae/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/pebble/PebbleIoThread.java#L170). The full exception reads `Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?`

**Comment by IzzySoft** [comment]
Just as a note: Confirmed here with Pebble Time Steel (latest FW) and GB 0.22.4 on Android 6 (Lineage). My details are in #889 (sorry for the dupe, missed this one on search).

Major detail to add here: tried with different watchfaces (TrekVolle, Healthify, TicToc), the crash is 100% reproducable (so it's unlikely to be due to a bug in the watchface's JS). Also 100% reproducable: as soon as background JS is turned off again, connect works without any issue.

**Comment by IzzySoft** [comment]
Any news on this? Need more logs or something?

**Comment by cpfeiffer** [comment]
I will investigate.

**Comment by exmosis** [comment]
For reference, I also found this was happening with my Pebble Time watch.

**Comment by VmKid** [comment]
Issue appears to be fixed as of 0.24.2.
